autopush_rs/src/util/megaphone.rs

```diff
@@ -0,0 +1,249 @@
+use std::collections::HashMap;
+use errors::Result;
+
+// A Service entry Key in a ServiceRegistry
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+struct ServiceKey(u32);
+
+// A list of services that a client is interested in and the last change seen
+#[derive(Debug)]
+pub struct ServiceList {
+    service_list: Vec<ServiceKey>,
+    change_count: u32,
+}
+
+#[derive(Debug)]
+struct ServiceRegistry {
+    lookup: HashMap<String, u32>,
+    table: Vec<String>,
+}
+
+// Return result of the first delta call for a client given a full list of service id's and versions
+#[derive(Debug)]
+pub struct ServiceClientInit {
```

Comment:
nitpick: tempted to kill this for a tuple struct since it's only used as one return type